### PR TITLE
New options to include/exclude directories through regular expressions

### PR DESCRIPTION
**Pull request by mlenkeit**
This PR addresses #33.

It's in draft state so far and adds an initial implementation for an `--include-pattern` option (incl. a test case for this option and for existing functionality). I will add an `--exclude-pattern` option that follows a similar pattern.

Depending on whether there are more advanced test cases required, I would suggest to refactor the test cases to create a *directory structure under test* e.g. by using the [tmp package](https://www.npmjs.com/package/tmp#synchronous-directory-creation).

@mateodelnorte what are your initial thoughts on this?

### CODE REVIEW

**Comment by mlenkeit** [comment]
How about testing this automatically on Travis rather than manually? We could install zsh on Travis (just tried it out, see [sample repo](https://github.com/mlenkeit/test-travis-ci/blob/0a54a6865795a419822c17c45abbc4a09762d364/.travis.yml#L1-L6)) and then run `child_process.exec` from the automated tests to invoke zsh. I would probably put this into a separate file (e.g. `test/zsh.js`) to separate this kind of test from regular unit tests.
Anyway, I'll also make sure to test it manually.

**Comment by mateodelnorte** [comment]
Automated bash and zsh tests would be fantastic. Thanks!

**Comment by mlenkeit** [comment]
I added a new `test/shells.js` file that runs tests for loop against sh, bash, and zsh. These tests now use a temporary directory with a couple of dummy subdirectories to test the functionality rather than the loop root directory itself. 

I also added the `--exclude-pattern` option following the same pattern as the `--include-pattern` option. It's indeed correct that in zsh, the regexp needs to be escaped/wrapped in quotes.

From my side, the PR would be ready for review and merging. Let me know if there's anything that prevents a merge.

**Comment by mlenkeit** [comment]
@mateodelnorte did you already have a chance to have a brief look at the proposed changes? Would be great if this could soon be merged, released and then make its way into `meta` 🙂

**Comment by patrickleet** [comment]
Links to new packages for convenience:
https://www.npmjs.com/package/which
https://www.npmjs.com/package/mkdirp
https://www.npmjs.com/package/tmp

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@ notifications:
   email: false
 
 before_install:
+  - sudo apt-get update
+  - sudo apt-get install -y zsh bash curl git
```
Comment (mateodelnorte) [inline .travis.yml]: are curl and git needed, here?

Comment (mlenkeit) [reply .travis.yml]: Thanks for pointing this out, I must have missed this. Fix it with 488da5e.

**Comment by mlenkeit** [comment]
@patrickleet I agree that the way multiple flags are currently handled is not ideal. However, I think especially considering that these flags can also be specified in `.looprc` and some might already have e.g. `include` and `includeOnly` in there, throwing an error would constitute an incompatible change, requiring to release a major version rather than a minor one.
I do think it's worth fixing, but I'd suggest tracking this separately.

**Comment by mateodelnorte** [comment]
LGTM

**Comment by mateodelnorte** [comment]
Thanks!

**Comment by mlenkeit** [comment]
Thanks for merging the PR. I just also closed the related issue.

@mateodelnorte would you mind creating a new release of `loop` that includes the new flags, so that renovate can make it also available in `meta`?

**Comment by mlenkeit** [comment]
Thanks for releasing it! 🎉

**Pull request merged**